src/couch_replicator_httpc_pool.erl

```diff
@@ -55,114 +58,113 @@ release_worker(Pool, Worker) ->
-        {empty, Waiting2} ->
-            NewCallers1 = NewCallers0,
-            Busy2 = State#state.busy -- [Worker],
-            Free2 = [Worker | State#state.free];
-        {{value, From}, Waiting2} ->
-            NewCallers1 = monitor_client(NewCallers0, Worker, From),
-            gen_server:reply(From, {ok, Worker}),
-            Busy2 = State#state.busy,
-            Free2 = State#state.free
-        end,
-        NewState = State#state{
-           busy = Busy2,
-           free = Free2,
-           waiting = Waiting2,
-           callers = NewCallers1
-        },
-        {noreply, NewState};
-   false ->
-        {noreply, State#state{callers = NewCallers0}}
-   end.
-
-handle_info({'EXIT', Pid, _Reason}, State) ->
-    #state{
-        url = Url,
-        busy = Busy,
-        free = Free,
-        waiting = Waiting,
-        callers = Callers
-    } = State,
-    NewCallers0 = demonitor_client(Callers, Pid),
-    case Free -- [Pid] of
-    Free ->
-        case Busy -- [Pid] of
-        Busy ->
-            {noreply, State#state{callers = NewCallers0}};
-        Busy2 ->
-            case queue:out(Waiting) of
-            {empty, _} ->
-                {noreply, State#state{busy = Busy2, callers = NewCallers0}};
-            {{value, From}, Waiting2} ->
-                {ok, Worker} = ibrowse:spawn_link_worker_process(Url),
-                NewCallers1 = monitor_client(NewCallers0, Worker, From),
-                gen_server:reply(From, {ok, Worker}),
-                NewState = State#state{
-                    busy = [Worker | Busy2],
-                    waiting = Waiting2,
-                    callers = NewCallers1
-                },
-                {noreply, NewState}
-            end
-        end;
-    Free2 ->
-        {noreply, State#state{free = Free2, callers = NewCallers0}}
-    end;
+handle_cast({release_worker, Worker}, State0) ->
+    State1 = release_worker_int(Worker, State0),
+    State2 = maybe_supply_worker(State1),
+    {noreply, State2}.
 
-handle_info({'DOWN', Ref, process, _, _}, #state{callers = Callers} = State) ->
+
+handle_info({'EXIT', Worker, _Reason}, State0) ->
+    State1 = release_worker_int(Worker, State0),
+    State2 = maybe_supply_worker(State1),
+    {noreply, State2};
+
+handle_info({'DOWN', Ref, process, _, _}, #state{callers = Callers} = State0) ->
     case lists:keysearch(Ref, 2, Callers) of
         {value, {Worker, Ref}} ->
-            handle_cast({release_worker, Worker}, State);
+            State1 = release_worker_int(Worker, State0),
+            State2 = maybe_supply_worker(State1),
+            {noreply, State2};
         false ->
-            {noreply, State}
-    end.
+            {noreply, State0}
+    end;
+
+handle_info(refresh_dns, State0) ->
+    #state{url=Url, dns_data=Data0}=State0,
+    State1 = case couch_replicator_dns:lookup(Url) of
+        {ok, {Data0, TTL}} ->
+            State0#state{dns_ttl=TTL};
+        {ok, {Data1, TTL}} ->
```

Comment:
You are just closing all connections down, without logging or re-opening them? That happens automatically?